os/fs/smartfs/smartfs_utils.c

```diff
@@ -1146,7 +1146,10 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 
 	/* Test if this is a request for the root directory */
 
-	if (*relpath == '\0') {
+	if (relpath == NULL) {
+		ret = -EINVAL;
+		goto errout;
```

Comment:
Do we need to go to the errout? How about returning here directly?
@Taejun-Kwon Please review this.

Reply:
prev code is proper. This means that user request something with directory '/mnt'
'mnt' is mount point, smartfs doesn't care path of mount point. hence '\0' is value of relpath of root directory.
Hence svace was wrong.